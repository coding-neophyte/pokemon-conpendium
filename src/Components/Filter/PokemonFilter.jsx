import React from 'react'

export default function PokemonFilter({ typeSelected, setSelectedType, pokeTypes }) {

    return (
        <div>
            <select aria-label='dropDown'onChange={(e) => setSelectedType(e.target.value)}>
                <option value='all'>All</option>
                {pokeTypes.map(poke => <option key={poke} value={poke}>{poke}</option>)}
            </select>
        </div>
    )
}
