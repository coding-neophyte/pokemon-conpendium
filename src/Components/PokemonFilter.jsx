import React from 'react'

export default function PokemonFilter({ typeSelected, setSelectedType, pokeTypes }) {

    return (
        <div>
            <select onChange={(e) => setSelectedType(e.target.value)}>
                <option value='all'>All</option>
                {pokeTypes.map(poke => <option key={poke} value={poke}>{poke}</option>)}
            </select>
        </div>
    )
}
