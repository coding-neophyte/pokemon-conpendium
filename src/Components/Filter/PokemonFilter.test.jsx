import { screen, render } from '@testing-library/react'
import PokemonFilter from './PokemonFilter'

it('should render filter component', () => {
    const types = ['pikachu', 'mew']
    render(<PokemonFilter pokeTypes={types} />)
    const dropDown = screen.getByLabelText(/dropDown/)
    expect(dropDown).toMatchSnapshot()
})
