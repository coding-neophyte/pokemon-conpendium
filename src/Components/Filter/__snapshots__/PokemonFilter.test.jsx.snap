// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render filter component 1`] = `
<select
  aria-label="dropDown"
>
  <option
    value="all"
  >
    All
  </option>
  <option
    value="pikachu"
  >
    pikachu
  </option>
  <option
    value="mew"
  >
    mew
  </option>
</select>
`;
